csharp/src/Drivers/BigQuery/BigQueryConnection.cs

```diff
@@ -337,33 +419,49 @@ private IArrowArray[] GetCatalogs(
             StringArray.Builder catalogNameBuilder = new StringArray.Builder();
             List<IArrowArray?> catalogDbSchemasValues = new List<IArrowArray?>();
             string catalogRegexp = PatternToRegEx(catalogPattern);
-            PagedEnumerable<ProjectList, CloudProject>? catalogs = this.client?.ListProjects();
+            PagedEnumerable<ProjectList, CloudProject>? catalogs;
+            List<string> projectIds = new List<string>();
 
-            if (catalogs != null)
+            try
             {
-                List<string> projectIds = catalogs.Select(x => x.ProjectId).ToList();
+                Func<Task<PagedEnumerable<ProjectList, CloudProject>?>> func = () => Task.Run(() =>
+                {
+                    // stick with this call because PagedAsyncEnumerable has different behaviors for selecting items
+                    return Client?.ListProjects();
+                });
 
-                if (this.includePublicProjectIds && !projectIds.Contains(publicProjectId))
-                    projectIds.Add(publicProjectId);
+                catalogs = ExecuteWithRetriesAsync<PagedEnumerable<ProjectList, CloudProject>?>(func).GetAwaiter().GetResult();
+
+                if (catalogs != null)
+                {
+                    projectIds = catalogs.Select(x => x.ProjectId).ToList();
+                }
+            }
+            catch
+            {
+                // TODO: Logging
+            }
```

Comment:
We weren't swallowing exceptions here before; why are we doing it now? Consider adding a comment with the justification.

Reply:
removed